server/src/main/java/com/linecorp/centraldogma/server/mirror/MirrorListener.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.server.mirror;
+
+import java.util.ServiceLoader;
+
+import javax.annotation.Nullable;
+
+/**
+ * A listener which is notified when a {@link Mirror} operation is started, completed or failed.
+ *
+ * <p>Implement this interface and register it using {@link ServiceLoader} to override the default behavior:
+ * <ul>
+ *     <li>{@link #onStart(Mirror)}: Logs an info message for the start of {@link Mirror}.</li>
+ *     <li>{@link #onComplete(Mirror, MirrorResult)}: Does nothing.</li>
+ *     <li>{@link #onError(Mirror, Throwable)}: Logs a warning message for the error of {@link Mirror}.</li>
+ * </ul>
+ */
+@Nullable
+public interface MirrorListener {
+
+    /**
+     * Invoked when the {@link Mirror} operation is started.
+     */
+    void onStart(Mirror mirror);
```

Comment:
Optional; If a `MirrorTask` is passed at `onStart` which contains a `mirrorId`, users may be able to differentiate the flow of a single `Mirror` across multiple callbacks via `mirrorId`. This may be useful since multiple mirrors may be run in a single leader with the introduction of zone-aware mirroring.

Reply:
`mirrorId` means the ID for a single mirroring task, not `Mirror#id()`, right?

> This may be useful since multiple mirrors may be run in a single leader with the introduction of zone-aware mirroring.

Sorry, I didn't understand this. Did you mean multiple mirrors can be run concurrently? Could you elaborate on this, please?